### PR TITLE
pt159338028 Feeback -Create API endpoints 

### PR DESCRIPTION
**Pull request by ayodejiojo**
**What does this PR do**
It tests the API endpoints and store data in an array

 **Description of Task to be completed**
Implemented the following  endpoint and tested it on postman
- POST /api/v1/signup: used to get the information and store in the array users.
- POST /api/v1/entry: used to get and add information and store in the array message.
-PUT/api/v1/:id: used to update entries.
-GET/api/v1/entries: used to get all entries made by the user.
-GET/api/v1/entries:id: used to get one entry done by the user
-POST/api/v1/signin: used to get information from the user and check if the user exists already in the array.

 **How should this be manually tested?**
The repo can be cloned and tested on postman.

**Any background context you want to provide**
none.

**What are the relevant pivotal tracker stories**
#159338028

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	const { title, message } = req.body;
+	if (title === ' ' || message === ' ' ) {
+		res.status(422).json('Please fill fields');
+	} 
+	 else if (title && message) {
+		database.entries.push(req.body);
+		res.status(201).json(database.entries);
+	} 
+	else {
+		res.status(400).json('Bad Request');
```
Comment (Celoka) [inline src/index.js]: This is a wrong status code, I believe you want to catch a server error. 500 should be appropriate.

Comment (Celoka) [reply src/index.js]: Change this wherever you are repeating yourself.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	if (firstname === ' ' || lastname === ' ' || email === ' ' || password === ' ') {
+		res.json('Please fill fields').status(422);
+	} else if (firstname && lastname && email && password) {
+		const user = database.users.filter(u => u.email === email && u.password === password);
+		if (user.length > 0 && user[0].email) {
+			res.status(409).json('Email and password already taken!');
+		} else {
+			database.users.push(req.body);
+			res.status(201).json(database.users);
+		}
```
Comment (Celoka) [inline src/index.js:51]: It is not ideal to log to the console. Again, your variable names should be in camel case.

File: src/index.js (hunk 1 above)
Comment (ayodejiojo) [reply src/index.js]: I have updated the server error to 500

File: src/index.js (hunk 2 above)
Comment (ayodejiojo) [reply src/index.js:51]: I have removed the console

**Pull request merged**